tests/test_main.py

```diff
@@ -1 +1,76 @@
-# write your code here
+from app.main import fill_tank
+
+
+def test_standard():
+    customer = {
+        "money": 3000,
+        "vehicle": {
+            "max_tank_capacity": 40,
+            "fuel_remains": 2,
+        }
+    }
+
+    fill_tank(customer, 1.33, 33)
+    assert customer["money"] == 2956.11
+    assert customer["vehicle"]["max_tank_capacity"] == 40
+    assert customer["vehicle"]["fuel_remains"] == 35
+
+
+def test_amount_none():
+    customer = {
+        "money": 3000,
+        "vehicle": {
+            "max_tank_capacity": 40,
+            "fuel_remains": 5,
+        }
+    }
+
+    fill_tank(customer, 1.33)
+    assert customer["money"] == 2953.45
+    assert customer["vehicle"]["max_tank_capacity"] == 40
+    assert customer["vehicle"]["fuel_remains"] == 40
+
+
+def test_amount_more_then_capacity():
+    customer = {
+        "money": 3000,
+        "vehicle": {
+            "max_tank_capacity": 40,
+            "fuel_remains": 5,
+        }
+    }
+
+    fill_tank(customer, 1.33, 50)
+    assert customer["money"] == 2953.45
+    assert customer["vehicle"]["max_tank_capacity"] == 40
+    assert customer["vehicle"]["fuel_remains"] == 40
+
+
+def test_money_not_enough():
+    customer = {
+        "money": 30,
+        "vehicle": {
+            "max_tank_capacity": 40,
+            "fuel_remains": 5,
+        }
+    }
+
+    fill_tank(customer, 4.65, 10)
+    assert customer["money"] == -0.23000000000000043
```

Comment:
How it's possible?)

Reply:
https://prnt.sc/26p0sdx (https://prnt.sc/26p0sxm)

Reply:
It's a bug. Check a new main, and rewrite a tests.